src/GridFieldNestedForm.php

```diff
@@ -0,0 +1,435 @@
+<?php
+
+namespace Symbiote\GridFieldExtensions;
+
+use Exception;
+use SilverStripe\Admin\ModelAdmin;
+use SilverStripe\Control\Controller;
+use SilverStripe\Control\Director;
+use SilverStripe\Control\HTTPResponse;
+use SilverStripe\Control\HTTPResponse_Exception;
+use SilverStripe\Core\Config\Configurable;
+use SilverStripe\Forms\GridField\AbstractGridFieldComponent;
+use SilverStripe\Forms\GridField\GridField;
+use SilverStripe\Forms\GridField\GridField_ColumnProvider;
+use SilverStripe\Forms\GridField\GridField_DataManipulator;
+use SilverStripe\Forms\GridField\GridField_HTMLProvider;
+use SilverStripe\Forms\GridField\GridField_SaveHandler;
+use SilverStripe\Forms\GridField\GridField_URLHandler;
+use SilverStripe\Forms\GridField\GridFieldDetailForm_ItemRequest;
+use SilverStripe\Forms\GridField\GridFieldStateAware;
+use SilverStripe\ORM\DataList;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\ORM\DataObjectInterface;
+use SilverStripe\ORM\Hierarchy\Hierarchy;
+use SilverStripe\ORM\SS_List;
+use SilverStripe\Versioned\Versioned;
+use SilverStripe\View\ViewableData;
+use Symbiote\GridFieldExtensions\GridFieldOrderableRows;
+
+/**
+ * Gridfield component for nesting GridFields
+ */
+class GridFieldNestedForm extends AbstractGridFieldComponent implements
+    GridField_URLHandler,
+    GridField_ColumnProvider,
+    GridField_SaveHandler,
+    GridField_HTMLProvider,
+    GridField_DataManipulator
+{
+    use Configurable, GridFieldStateAware;
+    
+    const POST_KEY = 'GridFieldNestedForm';
+
+    private static $allowed_actions = [
+        'handleNestedItem'
+    ];
+
+    private static $max_nesting_level = 10;
```

Comment:
```suggestion
    private static $default_max_nesting_level = 10;
```
This value is used as a default, so we should be explicit about that.

Please also add a PHPDoc for this since it will display in the API documentation. Same with any config you've added in this PR that isn't inherited.